spring-cloud-huawei-common/src/main/java/com/huaweicloud/common/transport/URLConfig.java

```diff
@@ -0,0 +1,77 @@
+package com.huaweicloud.common.transport;
+
+import com.huaweicloud.common.util.URLUtil;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
+
+/**
+ * @Author GuoYl123
+ * @Date 2019/12/19
+ **/
+public class URLConfig {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(URLConfig.class);
+
+  private List<String> urlList = new ArrayList<>();
+
+  private int index = 0;
+
+  private int resolveUrlSize = 0;
+
+  private int afterDnsResolveIndex = 0;
+
+  public String getUrl() {
+    if (resolveUrlSize > 0) {
+      String url = urlList.get(afterDnsResolveIndex);
+      //todo: need stick if success?
+      toggleDnsResolve();
+      return url;
+    }
+    return urlList.get(index);
+  }
+
+  public void setUrl(List<String> urls) {
```

Comment:
The logic seams to be "add Urls" not "set Urls"

Reply:
done